### ISSUE TITLE
JSON-RPC request without params violates specification with null

### ISSUE THREAD

**Issue by sim642** [eclipse/lsp4j#655]
When a JSON-RPC request without `params` (i.e. them being `null` in Java) is emitted, that gets written as `"params": null` instead of just omitting the `params` field from the output. The latter should be done instead.

## Specifications
[JSON-RPC specification](https://www.jsonrpc.org/specification#request_object) says:

> **params**
>    A _Structured value_ that holds the parameter values to be used during the invocation of the method. This member MAY be omitted. 

where "two structured types (Objects and Arrays)".

[LSP specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#requestMessage) says (in TypeScript syntax):

> ```
> /**
>  * The method's params.
>  */
> params?: array | object;
> ```

which exactly matches the JSON-RPC specification with the `?` meaning omission, but not `null` (which would have to be explicitly given as a `| null` variant).

### Implementation
The writing of `params` here:
https://github.com/eclipse/lsp4j/blob/22f8edce55379bd9d4ae1b96bc614e74549bc0f9/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapter.java#L404-L409
uses the `writeNullValue` function here:
https://github.com/eclipse/lsp4j/blob/22f8edce55379bd9d4ae1b96bc614e74549bc0f9/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapter.java#L449-L457
which goes out of its way to output a `null` instead of omitting missing `params`.

This clearly violates the specifications and trips up other JSON-RPC/LSP implementations that require incoming requests to follow the specification to the letter.

### Background
According to git blame, this behavior was introduced in #137, the goal of which was to fix the behavior for response `result` that indeed may not be omitted but may be `null`. Looks like the same logic was erroneously applied to request/notification `params` as well.

**Comment by jonahgraham** [comment]
I suspect to change this back (i.e. revert #137) we need to get xtext involved so to not regress all the xtext generated language servers. See https://github.com/eclipse/xtext-core/issues/558 for the discussion that led to #137's behaviour.

**Comment by cdietrich** [comment]
Is

Received message which is neither a response nor a notification message

fixed in viscose meanwhile?

**Comment by jonahgraham** [comment]
If this is still relevant a failing unit test or similar would be helpful so we can decide how to proceed

**Comment by pisv** [comment]
> I suspect to change this back (i.e. revert [#137](https://github.com/eclipse-lsp4j/lsp4j/pull/137)) we need to get xtext involved so to not regress all the xtext generated language servers. See [eclipse/xtext-core#558](https://github.com/eclipse/xtext-core/issues/558) for the discussion that led to [#137](https://github.com/eclipse-lsp4j/lsp4j/pull/137)'s behaviour.

I don't think that we need to revert all of #137 to fix this issue. The culprit really looks to be correctly summarized in the description background:

> According to git blame, this behavior was introduced in https://github.com/eclipse-lsp4j/lsp4j/pull/137, the goal of which was to fix the behavior for response result that indeed may not be omitted but may be null. Looks like the same logic was erroneously applied to request/notification params as well.

That is, we should not write `null` for null request/notification params, but should still do it for null response result.

This looks like a good candidate for 1.0.0.
